semgrep/semgrep/rule_match.py

```diff
@@ -6,6 +6,16 @@
 from typing import Optional
 
 from semgrep.pattern_match import PatternMatch
+from semgrep.semgrep_types import YAML_INTERNAL_KEYS
+
+
+def strip_internal_keys(obj: Any) -> Any:
+    if isinstance(obj, list):
+        return [strip_internal_keys(o) for o in obj]
+    elif isinstance(obj, dict):
+        return {k: v for k, v in obj.items() if k not in YAML_INTERNAL_KEYS}
```

Comment:
Were YAML internal keys being added to RuleMatch objects?

Reply:
Afaik RuleMatches are made from PatternMatch objects/semgrep-core output and this PR doesnt change what semgrep-core outputs?

Reply:
the metadata field from rules got the line numbers attached. (metadata is included in rule output)

Reply:
Oh I see